### ISSUE TITLE
After updating to FF2.7.1, updating Stack and deploying to Windows device, a device agent instance does not connect to Project Links.

### ISSUE THREAD

**Issue by SynoUser-NL** [FlowFuse/flowfuse#4367]
### Current Behavior

![image](https://github.com/user-attachments/assets/a18ba654-6b74-44e9-b9c8-7e79d8f2c7b1)

Notice the Project Nodes reconnecting instead of connected.

### Expected Behavior

After reverting to the previous deployment:
![image](https://github.com/user-attachments/assets/2d8ab1cc-4df4-408b-8e52-9aef2c80c04b)

### Steps To Reproduce

- have a NodeRED instance running on a Windows device using Project nodes from another instance in the same team
- update stack of the deployed instance to 2.7.1 version
- behaviour is as in screenshot above

### Environment

- FlowFuse version: 2.7.1
- Node.js version: 20.12.2
- Platform/OS: Windows (device agent updated to latest version)

### Have you provided an initial effort estimate for this issue?

I am not a FlowFuse team member

**Comment by knolleary** [comment]
@SynoUser-NL can you confirm what Node.js version you are using with the device agent? Is that 20.12.2 as well?

**Comment by SynoUser-NL** [comment]
Hi @knolleary , 
Yes, Node 20.12.2 is running on the Windows server running device agent.

**Comment by Steve-Mcl** [comment]
Good morning @SynoUser-NL 

Nick has assigned this to me and I am picking this up this morning. 

There are few more details I wish to know but it would be much quicker if we can set up a call - is that possible?  Additionally, if we could poke the bear it may provide valuable insight.

If not, then here are a few things that might help determine what is happing.

#### working agent
1. what were the details of the stack when device agent worked (node-red version)
2. what was the version of device-agent when working (same version?)
3. what was the version of project-nodes (check palette manager on the node-red)

### non working agent
1. what were the details of the stack when device agent does not work (node-red version)
2. what was the version of device-agent after update
3. what was the version of project-nodes (check palette manager on the node-red)

### Additional info
1. Does the windows server have any proxy environment variables set?
   1.  Start a `cmd` or `ps` terminal using the same accout that the device agent runs under, enter `set` and look for any env var that looks like `????_proxy`
1. Are you running the device agent as a service? (how does it start up?)

**Comment by Steve-Mcl** [comment]
Follow up info.

After a live meeting we determined this was related to proxy settings.  @SynoUser-NL environment uses a proxy for internet traffic and these are added to the parent instance of the device the user was having trouble with.

After updating to the latest versions of device-agent and project-nodes and adding `no_proxy` entry in the env vars, we got this  all working.

For clarification. The latest device-agent and project-nodes now have full standardised support for proxies and thus if set, require `no_proxy` if traffic is not intended to go via the proxy. 

This issue is resolved.
